ARMv7: Fix testStoreAfterClobber*

aoikonomopoulos opened the pull request:
#### 3663b1213e4bef8a1e1ce8723581d57e55a1ab60
<pre>
ARMv7: Fix testStoreAfterClobber*
<a href="https://bugs.webkit.org/show_bug.cgi?id=283968">https://bugs.webkit.org/show_bug.cgi?id=283968</a>

Reviewed by Justin Michaud.

Adjust for 64-bits and don&apos;t try to pin registers that have been removed
from the register set in Air::Code::Code().

* Source/JavaScriptCore/b3/testb3_8.cpp:
(testStoreAfterClobberExitsSideways):
(testStoreAfterClobberDifferentWidth):
(testStoreAfterClobberDifferentWidthSuccessor):
(testStoreAfterClobberExitsSidewaysSuccessor):

Canonical link: <a href="https://commits.webkit.org/288076@main">https://commits.webkit.org/288076@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bf63e83d48541351f70435814765ec9fa6309266

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81676 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1201 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35625 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86216 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32666 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1232 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9020 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63714 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21444 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84746 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/882 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74321 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44006 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/781 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28498 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31121 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/74655 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72179 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29096 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87655 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/80730 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/8913 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72056 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9096 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70146 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71291 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17787 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15369 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14279 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/103141 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/8865 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14400 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25052 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8705 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12226 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10514 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

justinmichaud commented on Source/JavaScriptCore/b3/testb3_8.cpp:
```diff
@@ -1094,6 +1094,10 @@ void testStoreAfterClobberExitsSideways()
     RegisterSetBuilder csrs;
     csrs.merge(RegisterSetBuilder::calleeSaveRegisters());
     csrs.exclude(RegisterSetBuilder::stackRegisters());
+#if CPU(ARM)
```
Let's leave a comment referencing the bug to undo this, see https://github.com/WebKit/WebKit/pull/37341/files#diff-cd055880f848bd628c629328b14251f6bce5ed9bd76ff64cf4236b75e4758b18R40

justinmichaud commented on Source/JavaScriptCore/b3/testb3_8.cpp:
```diff
@@ -1274,6 +1280,10 @@ void testStoreAfterClobberExitsSidewaysSuccessor()
     RegisterSetBuilder csrs;
     csrs.merge(RegisterSetBuilder::calleeSaveRegisters());
     csrs.exclude(RegisterSetBuilder::stackRegisters());
+#if CPU(ARM)
```
ditto

justinmichaud reviewed (approved):
r=me

justinmichaud reviewed (approved):
r=me

aoikonomopoulos committed:
ARMv7: Fix testStoreAfterClobber*
https://bugs.webkit.org/show_bug.cgi?id=283968

Reviewed by Justin Michaud.

Adjust for 64-bits and don't try to pin registers that have been removed
from the register set in Air::Code::Code().

* Source/JavaScriptCore/b3/testb3_8.cpp:
(testStoreAfterClobberExitsSideways):
(testStoreAfterClobberDifferentWidth):
(testStoreAfterClobberDifferentWidthSuccessor):
(testStoreAfterClobberExitsSidewaysSuccessor):

Canonical link: https://commits.webkit.org/288076@main
